JavaScript enhancements

angelsanzn opened the pull request:
JavaScript is a beautiful language and we all know it.

Commit:
Camel case!

Commit:
Get rid of moment.js dependency, and fight against js' Date API in the process.

Commit:
Inline everything that can be inlined in the exports of a module pattern, instead of creating a new function which just delegates to the proper one.

Commit:
Simplify check.

Commit:
Rewrite big if statement to a guard clause.

Commit:
Double equals is an evil wild beast which must be banished from Earth.

I only had to left out those occurences dealing with gMaps or jQuery, about which I'm not totally positive
that switching to triple equals won't break anything. Leave it up to you! ;)

Commit:
Inline most event handlers wherever it was possible and improved clarity.

ergl commented:
Hey, thanks for this!

Looks good and it even comes with witty commit messages - I'll take a closer look at it tomorrow and see if it all works nicely

Commit:
Use === everywhere.

The only missing places where those checks related
to the Google Maps API and XHR APIs.

Commit:
$.ajax: correct `datatype` to `dataType`.

Commit:
geocode.js: add missing vars.

Commit:
Add missing semicolons.
